net-misc/kea/kea-3.0.0.ebuild

```diff
@@ -0,0 +1,247 @@
+	test? ( dev-cpp/gtest )
+"
+RDEPEND="${COMMON_DEPEND}
+	acct-group/dhcp
+	acct-user/dhcp
+"
+BDEPEND="
+	>=dev-build/meson-1.8
+	doc? (
+		$(python_gen_any_dep '
+			dev-python/sphinx[${PYTHON_USEDEP}]
+			dev-python/sphinx-rtd-theme[${PYTHON_USEDEP}]
+		')
+	)
+	virtual/pkgconfig
+"
+
+python_check_deps() {
+	use doc || return 0;
+	python_has_version "dev-python/sphinx[${PYTHON_USEDEP}]" \
+		"dev-python/sphinx-rtd-theme[${PYTHON_USEDEP}]"
+}
+
+pkg_setup() {
+	if use doc || use shell; then
+		python_setup
+	fi
+}
+
+src_prepare() {
+	default
+
+	# Remove documentation from build
+	if use !doc; then
+		sed -e "/^subdir('doc')$/d" \
+		-i meson.build || die
+	fi
+
+	# Fix up doc path where examples go
+	sed -e "s:DATADIR / 'doc/kea':DATADIR / 'doc/${P}':" \
+		-i meson.build || die
+
+	# set shebang before meson if we are installing the shell
+	if use shell; then
+		sed -e 's:^#!@PYTHON@:#!/usr/bin/env python3:' \
+		-i src/bin/shell/kea-shell.in || die
+	fi
+
+	# Don't allow meson to install shell
+	sed -e 's:install\: true:install\: false:' \
+		-i src/bin/shell/meson.build || die
+
+	# do not create /run
+	sed -e '/^install_emptydir(RUNSTATEDIR)$/d' \
+		-i meson.build || die
+}
+
+src_configure() {
+	# https://bugs.gentoo.org/861617
+	# https://gitlab.isc.org/isc-projects/kea/-/issues/3946
+	#
+	# Kea Devs say no to LTO
+	filter-lto
+
+	local emesonargs=(
+		--localstatedir="${EPREFIX}/var"
+		-Drunstatedir="${EPREFIX}/run"
+		-Dkrb5=disabled
+		-Dnetconf=disabled
+		-Dcrypto=$(usex openssl openssl botan)
+		$(meson_feature mysql)
+		$(meson_feature postgres postgresql)
+		$(meson_feature test tests)
+	)
+	if use debug; then
+		emesonargs+=(
+			--debug
+		)
+	fi
+	meson_src_configure
+}
+
+src_test() {
+	# Get list of all test suites into an associative array
+	# the meson test --list returns either "kea / test_suite", "kea:shell-tests / test_suite" or "kea:python-tests / test_suite"
+	# Discard the shell tests as we can't run shell tests in sandbox
+
+	pushd "${BUILD_DIR}" || die
+	local -A TEST_SUITES
+	while IFS=" / " read -r subsystem test_suite ; do
+		if [[ ${subsystem} != "kea:shell-tests" ]]; then
+			TEST_SUITES["$test_suite"]=1
+		fi
+	done < <(meson test --list || die)
+	popd
+
+	# Some other tests will fail for interface access restrictions, we have to remove the test suites those tests
+	# belong to
+	local SKIP_TESTS=(
+		dhcp-radius-tests
+		kea-log-buffer_logger_test.sh
+		kea-log-console_test.sh
+		dhcp-lease-query-tests
+		kea-dhcp6-tests
+	)
+
+	if [[ $(tc-get-ptr-size) -eq 4 ]]; then
+		# see https://bugs.gentoo.org/958171 for reason for skipping these tests
+		SKIP_TESTS+=(
+			kea-util-tests
+			kea-dhcp-tests
+			kea-dhcp4-tests
+			kea-dhcpsrv-tests
+			dhcp-ha-lib-tests
+			kea-d2-tests
+		)
+	fi
+
+	for SKIP in ${SKIP_TESTS[@]}; do
+		unset TEST_SUITES["${SKIP}"]
+	done
+
+	meson_src_test ${!TEST_SUITES[@]}
+}
+
+install_shell() {
+	python_domodule src/bin/shell/*.py
+	python_doscript src/bin/shell/kea-shell
```

Comment:
This will fail because kea-shell (and the py files) are only generated in the BUILD_DIR, not the source dir.
It's extra annoying because foreach_impl overrided the build dir, so you'll need to do something like

ORIG_BUILD_DIR="${BUILD_DIR}" python_foreach_impl  ....

And then here use "${ORIG_BUILD_DIR}"/src/bin/...